x/rollup/keeper/msg_server_test.go

```diff
@@ -0,0 +1,236 @@
+		},
+		"successful message with multiple user deposit txs": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz, depositTxBz},
+			},
+			shouldError: false,
+		},
+		"invalid l1 attributes tx bytes": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{invalidTxBz, depositTxBz},
+			},
+			shouldError: true,
+		},
+		"non-deposit tx passed in as l1 attributes tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{cosmosEthTxBz, depositTxBz},
+			},
+			shouldError: true,
+		},
+		"user deposit tx passed in as l1 attributes tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{depositTxBz, depositTxBz},
+			},
+			shouldError: true,
+		},
+		"invalid user deposit tx bytes": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, invalidTxBz},
+			},
+			shouldError: true,
+		},
+		"non-deposit tx passed in as user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, cosmosEthTxBz},
+			},
+			shouldError: true,
+		},
+		"l1 attributes tx passed in as user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, l1AttributesTxBz},
+			},
+			shouldError: true,
+		},
+		"contract creation tx passed in as user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, contractCreationTxBz},
+			},
+			shouldError: true,
+		},
+		"one successful l1 user deposit tx and an invalid tx passed in as user deposit txs": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz, invalidTxBz},
+			},
+			shouldError: true,
+		},
+		"bank keeper mint coins failure": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz},
+			},
+			setupMocks: func() {
+				s.bankKeeper.EXPECT().MintCoins(gomock.Any(), types.ModuleName, gomock.Any()).Return(sdkerrors.ErrUnauthorized)
+			},
+			shouldError: true,
+		},
+		"bank keeper send coins failure": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz},
+			},
+			setupMocks: func() {
+				s.bankKeeper.EXPECT().SendCoinsFromModuleToAccount(gomock.Any(), types.ModuleName, gomock.Any(), gomock.Any()).Return(sdkerrors.ErrUnknownRequest)
+			},
+			shouldError: true,
+		},
+	}
+
+	for name, test := range tests {
+		s.Run(name, func() {
+			if test.setupMocks != nil {
+				test.setupMocks()
+			}
+			s.mockMintETH()
+
+			var resp *types.MsgApplyL1TxsResponse
+			resp, err = keeper.NewMsgServerImpl(s.rollupKeeper).ApplyL1Txs(s.ctx, test.msg)
+
+			if test.shouldError {
+				s.Require().Error(err)
+				s.Require().Nil(resp)
+			} else {
+				s.Require().NoError(err)
+				s.Require().NotNil(resp)
+
+				// TODO: Verify that the expected event types are emitted
+
+				// Verify that the l1 block info and l1 block history are saved to the store
+				expectedBlockInfo := eth.BlockToInfo(testutils.GenerateL1Block())
+				store := sdk.UnwrapSDKContext(s.ctx).KVStore(s.storeKey)
```

Comment:
Is this how the store is retrieved in other Cosmos SDK unit tests? Is the store ever kept on the keeper suite struct?

Reply:
Most cosmos sdk unit tests just use pre-existing query endpoints instead of accessing the store directly, however since we don't have any for asserting the l1 block info I opted to just check the store directly. I like your idea of keeping the store in the keeper suite so we can access it directly from the suite, I'll update that.

Reply:
Once we figure out https://github.com/polymerdao/monomer/issues/160, we'll probably add query endpoints. At that point, it would be better to use the query endpoints in tests than directly accessing the store.

This works for now.